pkg/controller/etcdproxy/etcdproxy_controller_test.go

```diff
@@ -5,21 +5,47 @@ import (
 	"reflect"
 	"testing"
 
-	"github.com/xmudrii/etcdproxy-controller/pkg/apis/etcd/v1alpha1"
 	"k8s.io/api/core/v1"
 	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/runtime"
+	kubeclient "k8s.io/client-go/kubernetes/fake"
+	dslisters "k8s.io/client-go/listers/apps/v1"
+	corelisters "k8s.io/client-go/listers/core/v1"
 	"k8s.io/client-go/tools/cache"
 	"k8s.io/client-go/tools/record"
 
+	"github.com/xmudrii/etcdproxy-controller/pkg/apis/etcd/v1alpha1"
 	etcdclient "github.com/xmudrii/etcdproxy-controller/pkg/client/clientset/versioned/fake"
 	etcdlisters "github.com/xmudrii/etcdproxy-controller/pkg/client/listers/etcd/v1alpha1"
-	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
-	kubeclient "k8s.io/client-go/kubernetes/fake"
-	dslisters "k8s.io/client-go/listers/apps/v1"
-	corelisters "k8s.io/client-go/listers/core/v1"
 )
 
+func newEtcdProxyControllerMock(config *EtcdProxyControllerConfig, startingEtcdStorages, startingObjects []runtime.Object) *EtcdProxyController {
+	indexer := cache.NewIndexer(cache.DeletionHandlingMetaNamespaceKeyFunc, cache.Indexers{})
```

Comment:
Using the same indexer for different types seems weird.  I'd create separate indexers.  You can still pass in a slice of runtime.Objects and use switch/casts